### PR TITLE
feature(monitoring): take screenshots with grafana image renderer

### PR DESCRIPTION
**Pull request by fruch**
- switch to grafana image renderer
- drop all of webdriver related implementation
- remove grafana snapshots - it's not in use for quite some time

## TODO:
- [x] - remove selenium from hydra image

### Testing

- [x] provision tests
- [x] test with ipv6: https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-10gb-3h-ipv6-test/8/

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

**Comment by fruch** [comment]
it's seems to be working, but one API keeps failing:
```
logger=context userId=0 orgId=1 uname= t=2024-05-28T20:36:19.657348907Z level=info msg="Request Completed" method=GET path=/api/user/orgs status=401 remote_addr=172.17.0.1 time_ms=1 duration=1.269817ms size=27 referer= handler=/api/user/orgs
```

it's adds one ugly banner at the top of the screenshot:
image

couldn't figure out why, @amnonh are you familiar with this ?

**Comment by fruch** [comment]
Provision tests didn't seem to generate the screenshot, needed to try manually to debug.

Prometheus snapshot on gce seems to be working with public address and failing  (retrying for 30min)

**Comment by fruch** [comment]
V2:

- added env variables to maximize the max height
- remove the width height definition from dashboards

**Comment by fruch** [comment]
a working example from provision test (it's not the full size of the dashboard, waiting for next run result that should be the full size):
https://cloudius-jenkins-test.s3.amazonaws.com/1883d5fb-24ee-4ced-a5ca-ffd85025de40/20240529_130043/grafana-screenshot-pr-7503-scylla-per-server-metrics-nemesis-20240529_130047-PR-provision-docker-PR-7503-monitor-node-1883d5fb-0.png 

grafana-screenshot-pr-7503-scylla-per-server-metrics-nemesis-20240529_130047-PR-provision-docker-PR-7503-monitor-node-1883d5fb-0 (1)

File: sdcm/cluster.py [hunk 1]
```diff
@@ -5616,6 +5615,8 @@ def start_scylla_monitoring(self, node):
             -d `realpath "{self.monitoring_data_dir}"` -l -v master,{self.monitoring_version} \
             -b "--web.enable-admin-api --storage.tsdb.retention.time={self.prometheus_retention}" \
             -c 'GF_USERS_DEFAULT_THEME=dark'
+            -D '--env RENDERING_VIEWPORT_MAX_HEIGHT=15000'
```
Comment (vponomaryov) [inline sdcm/cluster.py:5618]: Below you have following:

```
DEFAULT_SNAPSHOT_WIDTH = 1920
```

Shouldn't these be aligned?

**Review by vponomaryov — COMMENTED** [review]
LGTM if it works

File: sdcm/cluster.py [hunk 2]
```diff
@@ -5615,13 +5614,13 @@ def start_scylla_monitoring(self, node):
...
             PATH=$PATH:/usr/sbin ./start-all.sh \
-            -D "{labels}" \
+            -D "{labels} --env RENDERING_VIEWPORT_MAX_HEIGHT=15000 --env RENDERING_VIEWPORT_DEVICE_SCALE_FACTOR=4" \
             -s `realpath "{self.monitoring_conf_dir}/scylla_servers.yml"` \
             -n `realpath "{self.monitoring_conf_dir}/node_exporter_servers.yml"` \
             {scylla_manager_servers_arg} \
             -d `realpath "{self.monitoring_data_dir}"` -l -v master,{self.monitoring_version} \
             -b "--web.enable-admin-api --storage.tsdb.retention.time={self.prometheus_retention}" \
-            -c 'GF_USERS_DEFAULT_THEME=dark'
+            -c 'GF_USERS_DEFAULT_THEME=dark' \
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster.py:5623]: it's a leftover after add some extra arguments, but then moved the other place.

it's not needed, but it also didn't broke anything... (i.e. it's a bit like adding comma at the end of a multi line python list)

[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster.py]: removed it

**Comment by fruch** [comment]
working on all backends, and also on ipv6

improves log collect times from 9-10m to ~2.5m

**Review by vponomaryov — APPROVED** [review]
LGTM

**Review by soyacz — APPROVED** [review]
LGTM

**Comment by soyacz** [comment]
awesome change

**Comment by fruch** [comment]
> awesome change

Throwing away code is satisfying 

The sad part is it took me years to get to closing the loop on something I've asked years ago (thanks @amnonh)

**Pull request merged**